### ISSUE TITLE
Error when Running `vllm_inference.py`: `CancelledError()

### ISSUE THREAD

**Issue by chaosisnotrandomitisrhythmic** [modal-labs/modal-examples#763]
I have encountered an issue when attempting to run the `vllm_inference.py` script from the Modal Examples repository. Below are the steps I followed and the error I encountered:

### Steps to Reproduce
1. Download the script using curl:
   ```bash
   curl -o vllm_inference.py https://raw.githubusercontent.com/modal-labs/modal-examples/main/06_gpu_and_ml/llm-serving/vllm_inference.py
   ```
2. Set up a virtual environment and activate it:
   ```bash
   python3 -m venv .venv
   source .venv/bin/activate
   ```
3. Install `modal` and run setup:
   ```bash
   pip install modal
   modal setup
   ```
4. Run the script:
   ```bash
   modal run vllm_inference.py
   ```

### Error Encountered
```bash
Stopping app - uncaught exception raised locally: CancelledError().
```

### Environment
- **Operating System**: macOS
- **Python Version**: 3.11.4

### Checks Performed
- [x] The `HF_TOKEN` exists both locally and as a Modal Secret in the dashboard with a valid token value.
- [x] Access to the gated model [mistralai/Mistral-7B-Instruct-v0.2](https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.2) has been granted.

### Additional Context
I have been very impressed with Modal so far, exceptional work! Aside from resolving this issue, could you also provide guidance on how to debug this type of error? I am looking to learn more about effective troubleshooting techniques in such scenarios. 🧙‍♂️✨🔮

Thank you for your assistance!

**Comment by charlesfrye** [comment]
Thanks for the kind words! Will take a look at this issue.

> could you also provide guidance on how to debug this type of error?

Try running with the environment variable `MODAL_LOGLEVEL` set to `debug` for additional output.

**Comment by chaosisnotrandomitisrhythmic** [comment]
```bash
(.venv) ➜  export MODAL_LOGLEVEL=debug
(.venv) ➜  modal run vllm_inference.py
[Thread-1 (thread_inner)] 2024-06-06T18:46:21+0200 Connecting to api.modal.com using scheme https
[Thread-1 (thread_inner)] 2024-06-06T18:46:21+0200 Client: Starting
[Thread-1 (thread_inner)] 2024-06-06T18:46:21+0200 Sending request to /modal.client.ModalClient/ClientHello
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/AppCreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Created new app with id ap-1iKPJKxLTQZqYgO5nTZOU5
✓ Initialized. View run at https://modal.com/chaosisnotrandomitisrhythmic/main/apps/ap-1iKPJKxLTQZqYgO5nTZOU5
⠋ Creating objects...[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Starting infinite loop _run_app.<locals>.<lambda>
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/AppGetLogs
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/FunctionPrecreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/AppHeartbeat
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/MountGetOrCreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/SecretGetOrCreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/MountGetOrCreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Computing checksums for 1 files using 12 workers
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/ImageGetOrCreate
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/MountPutFile
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending app disconnect/stop request
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Sending request to /modal.client.ModalClient/AppClientDisconnect
Stopping app - uncaught exception raised locally: CancelledError().
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 App logs are done
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Logging exited gracefully
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 App disconnected
[Thread-1 (thread_inner)] 2024-06-06T18:46:22+0200 Exiting infinite loop for _run_app.<locals>.<lambda>
╭─ Error 
```

**Comment by charlesfrye** [comment]
Ah, I suspect that this is because you just copied the single file instead of the entire repo. Could you check whether the error occurs when you clone the repo and run the command from the root --  think that'd be `MODAL_LOGLEVEL=debug modal run 06_gpu_and_ml/llm-serving/vllm_inference.py`

**Comment by chaosisnotrandomitisrhythmic** [comment]
I followed your suggestion and ran the script from the repository's root directory with the debug log level set, but the issue persisted. However, I've got some good news—I managed to solve the problem! The root cause was a mismatch between the local and remote secret key names [here](https://github.com/modal-labs/modal-examples/blob/0b252dc8df6380ed11556b8f7a42736b99bce317/06_gpu_and_ml/llm-serving/vllm_inference.py#L82) `"huggingface-secret" != "huggingface"`

After correcting this, everything started working smoothly without the need to clone the full repository.

My apologies for not catching this discrepancy sooner. A more specific error message, like `Could not find secret: 'huggingface-secret'`, might help make such issues more apparent and fool-proof in the future. 😄

Thanks for your support and patience!
